lib/streamlit/elements/number_input.py

```diff
@@ -289,13 +289,20 @@ def _number_input(
         # Ensure that the value matches arguments' types.
         all_ints = int_value and int_args
 
-        if (min_value and min_value > value) or (max_value and max_value < value):
-            raise StreamlitAPIException(
-                "The default `value` of %(value)s "
-                "must lie between the `min_value` of %(min)s "
-                "and the `max_value` of %(max)s, inclusively."
-                % {"value": value, "min": min_value, "max": max_value}
-            )
+        if min_value is not None and value is not None:
```

Comment:
Specifically written like this to avoid type errors. Not sure if there's a better way as I tried:
```
if min_value is not None and if min_value > value:
```

Reply:
Yeah, this is good! One nit: can we combine all three if-clauses together, rather than having a nested if right below this one:

```python
if min_value is not None and value is not None and min_value > value:
    raise StreamlitAPIException(...)
```